moduleframework/tools/dockerlint.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+#
+# Meta test family (MTF) is a tool to test components of a modular Fedora:
+# https://docs.pagure.org/modularity/
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# he Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Jan Scotka <jscotka@redhat.com>
+#
+from __future__ import print_function
+import os
+
+from moduleframework import module_framework
+from moduleframework import dockerlinter
+from moduleframework.avocado_testers import container_avocado_test
+
+
+class DockerFileLinter(module_framework.AvocadoTest):
+    """
+    :avocado: enable
+
+    """
+
+    dp = None
+
+    def setUp(self):
+        # it is not intended just for docker, but just docker packages are
+        # actually properly signed
+        self.dp = dockerlinter.DockerfileLinter()
+        if self.dp.dockerfile is None:
+            self.skip()
+
+    def _test_for_env_and_label(self, docker_env, docker_label, env=True):
+        label_found = True
+        if env:
+            label = self.dp.get_docker_specific_env(docker_env)
+        else:
+            label = self.dp.get_specific_label(docker_env)
+        if not label:
+            label_found = self.dp.get_specific_label(docker_label)
+        return label_found
+
+    def test_architecture_in_env_and_label_exists(self):
+        self.assertTrue(self._test_for_env_and_label("ARCH=", "architecture"))
```

Comment:
Can you please point me to guidelines which require dockerfiles to have environment variable `ARCH`?

Reply:
see here: https://fedoraproject.org/wiki/Container:Guidelines#LABELS architecture is required.
But I got your point. We can only test if LABEL architecture is present.
My test was based on container-image-template: https://github.com/container-images/container-image-template/blob/master/Dockerfile.template#L5
There are mentioned as ARCH and architecture.
But I will fix it.

Reply:
You are right, it's in there. I missed it, sorry.

Therefore we should check the presence of env vars and maybe produce a warning it they are missing.